internal/coreinternal/errorutil/http.go

```diff
@@ -0,0 +1,23 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package errorutil // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/errorutil"
+
+import (
+	"net/http"
+
+	"go.opentelemetry.io/collector/consumer/consumererror"
+)
+
+func HTTPError(w http.ResponseWriter, err error) {
```

Comment:
are you really going to reuse that code outside of sapmreceiver?

Reply:
yes. look at this PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35327.

It's for loki receiver.